### PR TITLE
Add support for more types in gather op.

### PR DESCRIPTION
**Pull request by tsocha**
Add suport for types:
-i8
-i16
-i32
-i64
-u8
-u16
-u32
-u64

### CODE REVIEW

File: src/ngraph/runtime/cpu/builder/gather.cpp [hunk 1]
```diff
@@ -29,116 +29,126 @@ namespace ngraph
...
+                {
+                    functor = prepare_functor<uint32_t>(node, args, out, external_function);
+                }
+                else if (element_type == element::u64)
+                {
+                    functor = prepare_functor<uint64_t>(node, args, out, external_function);
+                }
+                else if (element_type == element::boolean)
+                {
+                    functor = prepare_functor<bool>(node, args, out, external_function);
```
Comment (rkimballn1) [inline src/ngraph/runtime/cpu/builder/gather.cpp]: in ngraph element::boolean is not a c type bool, but is a `char` type.
Would be nice to have a unit test to check this out. Really does need a unit test for all types.

Comment (tsocha) [reply src/ngraph/runtime/cpu/builder/gather.cpp]: OK. I will add these tests and change c type to char.

Comment (rkimballn1) [reply src/ngraph/runtime/cpu/builder/gather.cpp]: It still looks like it is type bool

Comment (tsocha) [reply src/ngraph/runtime/cpu/builder/gather.cpp]: Sorry, I missed that file in git add.

**Pull request merged**